server.js

```diff
@@ -21,7 +23,86 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  // console.log("req", req)
+  // console.log("res", res)
+  // res.send({responseMessage: "Hello Technigo!"}); 
+  res.json({responseMessage: "Hello Technigo!"});
+});
+// HTMLElement.addEventListener('nameOfTheListener', () => {
+
+// })
+
+// EP1 Technigo members code along
+app.get("/members", (request, response) => {
+  const { name, role } = request.query
+  let members = technigoMembers
+
+  if (role) {
+    members = members.filter(singleTechnigoMember => singleTechnigoMember.role.toLowerCase() ===
+    role.toLowerCase())
+    // members = technigoMembers.filter(singleTechnigoMember => { return singleTechnigoMember.role === role}) //this works too
+  }
+  if (name) {
+    members = members.filter(singleTechnigoMember => { return singleTechnigoMember.name.toLowerCase() === name.toLowerCase()})
+  }
+
+  response.status(200).json({
+    success: true,
+    message: "OK",
+    body: {
+      technigoMembers: members
+    }
+  });
+});
+
+app.get("/members/:id", (request, response) => {
+  const singleMember = technigoMembers.find((member) => {
+    return member.id === +request.params.id
+    // return member.id === Number(request.params.id) //this works too
+    // return member.id.toString() === request.params.id // and this
+  })
+  if (singleMember) {
+    response.status(200).json({
+      success: true,
+      message: "OK",
+      body: {
+        member: singleMember
+      }
+    })
+  } else {
+    response.status(404).json({
+      success: false,
+      message: "Not found",
+      body: {}
+    })
+  }
+
+  console.log(singleMember)
+  
+});
+
+// EP2 Books
+
+app.get("/books", (request, response) => {
+  const { title, authors } = request.query
+  let books = booksData
+
+  if (title) {
+    books = books.filter(singleBookData => singleBookData.title.toLowerCase() ===
+    title.toLowerCase())
+  }
+
+  if (authors) {
+    books = books.filter(singleBookData => singleBookData.authors.toLowerCase() === authors.toLowerCase())
+  }
```

Comment:
Nicely done with the if statements and filtering here of title and author. It works fine with queries.